packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java

```diff
@@ -70,5 +78,27 @@ public void execute() throws ExecutionFailedException {
             throw new ExecutionFailedException(
                     "Failed to run TypeScript endpoint generator", e);
         }
+        // This is a hack as Hilla does not have any other way to contribute to
+        // the generated vaadin.ts
+        try {
+            File vaadinTs = new File(outputDirectory,
+                    FrontendUtils.BOOTSTRAP_FILE_NAME);
+            if (!productionMode && vaadinTs.exists()) {
+                String current = IOUtils.toString(vaadinTs.toURI(),
+                        StandardCharsets.UTF_8);
+                current += """
+                        //@ts-ignore
+                        if (import.meta.env.DEV) {
+                            import("Frontend/generated/jar-resources/dev-tools-database.js");
+                        }
+                        """;
+                FileUtils.writeStringToFile(vaadinTs, current,
+                        StandardCharsets.UTF_8);
+            }
+        } catch (IOException e) {
+            throw new ExecutionFailedException("Failed to update vaadin.ts", e);
+
+        }
+
```

Comment:
Couldn't an `if (productionMode) { return; }` prevent opening and reading `vaadinTs` when it is building for production?

Reply:
Is that better?

Reply:
The current version does not read vaadinTs in production mode either

Reply:
Extracted to a method to make it cleaner